### PR TITLE
174294761 | Set guest_token cookie for guest orders

### PR DESCRIPTION
**Pull request by derrickp**
Pivotal 174294761 -> https://www.pivotaltracker.com/story/show/174294761 

Something is blowing away this cookie through all of the redirects that are happening and a new one gets assigned.

So by restoring this cookie for guest orders we can ensure that the user is not redirected to the login page for their cookie guest token not matching the one on the order.

Since we're getting the order from the MD5 hash that came back from the redirect, I think we can be fairly sure that they should have the guest_token from that order.

### CODE REVIEW

**Comment by cjreeve** [comment]
@derrickp sorry and hands up that I left our Adyen fork in a bit of a mess. Note that we are currently using an older ref [057647c](https://github.com/watg/solidus-adyen/commit/057647ca2995cb38ba38b38359d3aa7af7305c62) because my later attempts to resolve [this error](https://sentry.io/share/issue/e3752a942bf24cbfa589d3c799984950/) resulted just a 401 error, which had a worse effect of kicking the customer back to their basket page.

I'm not sure the best way to clean this up. Perhaps after you have merged your fix we could pix/reject commits from the master branch so that that my unwanted commits in between are removed. However, I don't think we are configured to allow force pushes into master.

**Comment by derrickp** [comment]
@cjreeve Which PR was it that merged the broken code into master?

**Comment by cjreeve** [comment]
@derrickp the last two pull requests.

First this: https://github.com/watg/solidus-adyen/pull/3

What this pull request did was remove any check as to whether the payment was in an expected/correct state.

Then I changed my mind about how I should best implement it but it (as far as I remember) it does the same thing: https://github.com/watg/solidus-adyen/pull/4

**A bit more background**:
These changes resulted in the failed payment to be found but it then attempted to push it to the failed state again. It had already been marked as failed elsewhere but not assigned the reference. An alternative solution could be to keep my changes and then add an extra check to ensure a payment does not attempt to be marked to the same state it is already in. However, ideally the payment should be given its reference as soon as we hear back from Adyen.

I never got the gem tests running on my machine.

I was asked to stop working on this so it could be resolved as part of the Rowan project.

**Review by jarednorman — APPROVED** [review]

**Comment by jarednorman** [comment]
Approved, pending sorting out the git history as discussed above.

**Comment by derrickp** [comment]
@cjreeve Sorry, just getting back to this now. How would you like to handle the rewind of master? Would you rather just revert those commits or actually fully rewind?

**Comment by cjreeve** [comment]
@derrickp yes I can do this. I don't know what is best in this case. I'm not sure if there is an argument for rolling back two pull request changes and pretending they never happened. My personal preference for my own record would be to revert my changes as a new commit.

**Review by MadelineCollier — APPROVED** [review]

**Pull request merged**
